test/hotspot/jtreg/runtime/8176717/TestInheritFD.java

```diff
@@ -238,10 +249,13 @@ public static void main(String[] args) throws Exception {
         pb.redirectOutput(commFile); // use temp file to communicate between processes
         pb.start();
 
-        if (waitForSubPids(commFile)) {
+        Result result = waitForSubPids(commFile);
+        if (result == Result.FOUND_RETAINS_FD) {
             System.out.println("Log file was not inherited by third VM.");
-        } else {
+        } else if (result == Result.FOUND_LEAKS_FD) {
             throw new RuntimeException("Log file was leaked to the third VM.");
+        } else {
+            throw new RuntimeException("Found neither message, I am confused!");
```

Comment:
I need some help here, should this really throw? I assume that that makes the test fail, and maybe that's not actually what we want?

Reply:
I think a fail is all right as long as we can distinguish between the cases.

Reply:
No need for the "I am confused" part. This case may be unexpected but if we reach it then it is an error.

Reply:
I understood it as Johan wanted the exception to be reported as an "Error" instead of a "Fail" to show that the testcase is bad and not (necessarily) the JVM.

Reply:
Yes, what Leo said. Some way of signalling to the CI/CD system that it's the test that didn't function correctly, not that the test found an issue. Seems like that doesn't exist, so differentiating the two cases with a log line is good enough.

Reply:
If the test terminates by throwing an exception then the test failed. We often have cases in a test where we should not reach and it is normal to place a throw there in case we ever do - similar to VM `ShouldNotReachHere()`. The fact it may indicate a test issue rather than a VM issue does not matter. Comments in "log" lines don't help because they rarely get read.